### ISSUE TITLE
Segmentation fault: 11 when convert tflite model

### ISSUE THREAD

**Issue by DeqingSun** [kendryte/nncase#119]
I was trying to convert a tensorflow sample mnist model and I got Segmentation fault: 11 on mac and app crash on Windows. I was using NNCase v0.2.0 Beta3.

I used following code to export tflite

~~~
# refer to https://github.com/frogermcs/MNIST-TFLite
model.summary()

# Reset metrics before saving so that loaded model has same state,
# since metric states are not preserved by Model.save_weights
model.reset_metrics()

tflite_mnist_model = loadPath+"mnist_model.tflite"

# Get the concrete function from the Keras model.
run_model = tf.function(lambda x : model(x))

# Save the concrete function.
concrete_func = run_model.get_concrete_function(
    tf.TensorSpec(model.inputs[0].shape, model.inputs[0].dtype)
)

tf.saved_model.save(model, loadPath, concrete_func)

# Convert the model.
converter = tf.lite.TFLiteConverter.from_saved_model(loadPath)
tflite_model = converter.convert()

open(tflite_mnist_model, "wb").write(tflite_model)
~~~

and inputs outputs are

~~~
== Input details ==
name: x
shape: [ 1 28 28]
type: <class 'numpy.float32'>

== Output details ==
name: Identity
shape: [ 1 10]
type: <class 'numpy.float32'>
~~~

tflite model is attached.

[mnist_model.tflite.zip](https://github.com/kendryte/nncase/files/4527055/mnist_model.tflite.zip)

**Comment by sunnycase** [comment]
Input should be a 4D tensor

**Comment by DeqingSun** [comment]
Thanks for the reply. I changed keras setting to a 28*28*1 shape
~~~
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])
~~~
And now inputs outputs are
~~~
== Input details ==
name: x
shape: [ 1 28 28  1]
type: <class 'numpy.float32'>

== Output details ==
name: Identity
shape: [ 1 10]
type: <class 'numpy.float32'>
~~~

But the error is still the same.
I've attached new tflite file
[mnist_model4d.tflite.zip](https://github.com/kendryte/nncase/files/4537354/mnist_model4d.tflite.zip)

**Comment by krishnak** [comment]
NCHW format?

**Comment by DeqingSun** [comment]
Still crash

~~~
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(1, 28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])
~~~


~~~
== Input details ==
name: x
shape: [ 1  1 28 28]
type: <class 'numpy.float32'>

== Output details ==
name: Identity
shape: [ 1 10]
type: <class 'numpy.float32'>
~~~

[mnist_model4d.tflite.zip](https://github.com/kendryte/nncase/files/4543255/mnist_model4d.tflite.zip)

**Comment by krishnak** [comment]
the issue with your tflite file and not nncase,

1 can you do a keras

print(model.summary())

2. how are you saving the keras file before you convert it to tensorflow lite.
3. what command are you using to convert keras to tensorflow lite

**Comment by krishnak** [comment]
sorry you have provided the details in the above message. You are using

# Convert the model.
converter = tf.lite.TFLiteConverter.from_saved_model(loadPath)
tflite_model = converter.convert()

open(tflite_mnist_model, "wb").write(tflite_model)

this will not work

use stand alone application using the following command

the following works in keras 2/tensorflow 2

save keras file like this 
model.save("yourmodel.h5")

   tflite_convert --output yourmodel.tflite --keras_model_file yourmodel.h5

**Comment by sunnycase** [comment]
> 
> 
> Thanks for the reply. I changed keras setting to a 28x28x1 shape
> 
> ```
> model = tf.keras.models.Sequential([
>   tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
>   tf.keras.layers.Dense(128, activation='relu'),
>   tf.keras.layers.Dropout(0.2),
>   tf.keras.layers.Dense(10, activation='softmax')
> ])
> ```
> 
> And now inputs outputs are
> 
> ```
> == Input details ==
> name: x
> shape: [ 1 28 28  1]
> type: <class 'numpy.float32'>
> 
> == Output details ==
> name: Identity
> shape: [ 1 10]
> type: <class 'numpy.float32'>
> ```
> 
> But the error is still the same.
> I've attached new tflite file
> [mnist_model4d.tflite.zip](https://github.com/kendryte/nncase/files/4537354/mnist_model4d.tflite.zip)

I did't encounter an issue with this model. Did you provide 28x28 images as dataset?

**Comment by sunnycase** [comment]
Close as inactive.

**Comment by Narduzzi** [comment]
Hello, 

I am facing the same issue. I have created a model in Keras:

```python3
# create network
import tensorflow as tf

inputs = tf.keras.layers.Input((28,28,1))
x = tf.keras.layers.Conv2D(32,5,activation="relu")(inputs)
f = tf.keras.layers.Flatten()(x)
out = tf.keras.layers.Dense(10, activation="softmax")(f)
model = tf.keras.models.Model(inputs=inputs, outputs=out)
model.compile(optimizer=tf.keras.optimizers.Adam(learning_rate=1e-3), loss="categorical_crossentropy")
model.summary()

model.save("model.h5")
```
and converted it using TFLite:

```bash
tflite_convert --keras_model_file model.h5 --output model_conv.tflite
```

When I try to convert it in kmodel using ncc, ncc crashes.

```bash
ncc compile model_conv.tflite model.kmodel -i tflite

1. Import graph...
Segmentation fault (core dumped)
```

Configuration:
 - Tensorflow 2.5.0
 - nncase 0.2.0-beta4
 - Ubuntu 20.04LTS
 
 Any hint about the source of the error?
